File: src/main/java/com/gmm/modules/stock/StockCreate.java
package com.gmm.modules.stock;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;

import java.math.BigDecimal;
import java.time.LocalDate;

@Builder
public record StockCreate(
        @NotBlank(message = "티커를 입력해주세요.")
        String ticker,
        int quantity,
        @NotNull(message = "주문 금액을 입력해주세요.")
        BigDecimal orderPrice,
        @NotNull(message = "주문일을 입력해주세요.")
        LocalDate orderDate
) {

    public void validate() {

    }
}

File: front/src/entity/stock/StockCreate.ts
import { format } from 'date-fns'

export default class StockCreate {
  public readonly ticker: string
  public readonly quantity: number | null
  public readonly orderPrice: number | null
  public readonly orderDate: string

  constructor(
    ticker: string = '',
    quantity: number | null = null,
    orderPrice: number | null = null,
    orderDate: string = format(new Date(), 'yyyy-MM-dd')
  ) {
    this.ticker = ticker
    this.quantity = quantity
    this.orderPrice = orderPrice
    this.orderDate = orderDate
  }

}
